pyomo/core/expr/numeric_expr.py

```diff
@@ -975,8 +975,22 @@ def _to_string(self, values, verbose, smap, compute_values):
             return "{0} {1}".format(values[0],values[1])
         return "{0} + {1}".format(values[0],values[1])
 
-    def is_potentially_variable(self):
-        return False
+    def create_node_with_local_data(self, args, classtype=None):
+        assert classtype is None
+        try:
+            npv_args = all(
+                type(arg) in native_types or not arg.is_potentially_variable()
+                for arg in args
+            )
+        except AttributeError:
+            # We can hit this during expression replacement when the new
+            # type is not a PyomoObject type, but is not in the
+            # native_types set.  We will play it safe and clear the NPV flag
+            npv_args = False
```

Comment:
I understand the logic of this, but throughout the expression code we assume that (apart from in the `generate_*` methods) if the argument is not in `native_types`, then it adheres to the `NumericValue` API.  Instead of trapping for this here, would it be better to process the argument to the expression replacement visitor method?

Reply:
I honestly just copied this from the `NPV_Mixin` class. I am certainly fine with this change. Should I change the implementation in `NPV_Mixin`?

Reply:
Ahh yes.  I remember that logic now.  I vote we leave this as-is, although it might be worthwhile to add a note that in a future rework of the `SumExpression` internals we should reorder the class hierarchy to remove the need for that override.